taskfile/read/taskfile.go

```diff
@@ -121,7 +121,15 @@ func Taskfile(readerNode *ReaderNode) (*taskfile.Taskfile, error) {
 			return err
 		}
 
-		if v >= 3.0 && len(includedTaskfile.Dotenv) > 0 {
+		tasksWithDotenv := false
+		for _, v := range includedTaskfile.Tasks {
+			if len(v.Dotenv) > 0 {
+				tasksWithDotenv = true
+				break
+			}
+		}
+
+		if v >= 3.0 && (len(includedTaskfile.Dotenv) > 0 || tasksWithDotenv) {
```

Comment:
Hey @pd93,

Regarding your question on this...

I believe that the reason I added this validation at the time was because, if we allowed dotenv on included Taskfiles, this would cause an unintended side-effect how things work today on the code. Basically the env in the included Taskfile would be _merged_ into the main Taskfile. Given that:

1) I believe you can revert your changes in this validation here and keep the way it works on master
2) In a future PR, we could allow dotenv on included Taskfiles, but that would require a refactor so these env would be _scoped_ in the included Taskfile and not merged into the main Taskfile. In short, dotenv would only affect tasks on its own Taskfile and not all tasks. At least that's how I would expect it to work, but this is open to discussion, and we may want to open an issue about this.

Reply:
I suspect I'm misunderstanding slightly, but these sound contradictory to me?

> I don't think we can undo the changes you made on the validation

> you can revert your changes in this validation here and keep the way it works on master

Could you please clarify for me. Would you like to keep the changes in 4614c77 or revert them?

Reply:
Hey @pd93,

Sorry for the confusion.

I meant keep the same behavior as master. Taskfiles with _global_ dotenvs would be problematic (until we refactor to _scope_ them to be valid for tasks in that specific Taskfile only). Tasks with local dotenvs are fine, because they are already scoped by design.

I just reverted your last commit and merged to master. Thanks! 🙂